TESTS/mbedmicro-rtos-mbed/timer/main.cpp

```diff
@@ -0,0 +1,752 @@
+/*
+ * Copyright (c) 2017, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "rtos.h"
+#include "hal/us_ticker_api.h"
+
+using namespace utest::v1;
+
+extern uint32_t SystemCoreClock;
+
+/* Macro to define delta based on CPU clock frequency.
+ *
+ * Note that some extra time is counted by the timer.
+ * Additional time is caused by the function calls and
+ * additional operations performed by wait and
+ * stop functions before in fact timer is stopped. This may
+ * add additional time to the counted result.
+ *
+ * To take in to account this extra time we introduce DELTA
+ * value based on CPU clock (speed):
+ *   DELTA = TOLERANCE_FACTOR / SystemCoreClock * US_FACTOR
+ *
+ *   e.g.
+ *   For K64F          DELTA = (10000 / 120000000) * 1000000 = 83[us]
+ *   For NUCLEO_F070RB DELTA = (10000 /  48000000) * 1000000 = 208[us]
+ *   For NRF51_DK      DELTA = (10000 /  16000000) * 1000000 = 625[us]
+ */
+#define US_PER_SEC       1000000
+#define TOLERANCE_FACTOR 10000.0f
+#define US_FACTOR        1000000.0f
+
+static const int delta_sys_clk_us = ((int) (TOLERANCE_FACTOR / (float)SystemCoreClock * US_FACTOR));
+
+#define DELTA_US delta_sys_clk_us
+#define DELTA_S  ((float)delta_sys_clk_us/US_PER_SEC)
+
+static Timer *p_timer;
```

Comment:
Please initialize p_timer to NULL so the test in timer_user_ticker_setup_handler is valid.